src/ChartTemplates/StackChart/StackChartTemplate.js

```diff
@@ -1,41 +1,100 @@
+/* eslint-disable camelcase */
 import React from 'react';
 import propTypes from 'prop-types';
-import { Chart, ChartBar, ChartAxis, ChartStack, ChartThemeColor, ChartTooltip } from '@patternfly/react-charts';
+import { Chart, ChartLegend, ChartBar, ChartAxis, ChartStack, ChartTooltip } from '@patternfly/react-charts';
+import { global_palette_gold_300, global_palette_gold_400, global_palette_orange_300, global_palette_red_200 } from '@patternfly/react-tokens';
 
-export const StackChart = ({ ...props }) => {
-    return (
-        <Chart
-            ariaDesc={ props.ariaDesc }
-            ariaTitle={ props.ariaTitle }
-            legendData={ props.legendData }
-            legendPosition={ props.legendPosition }
-            height={ props.height }
-            padding={ props.padding }
-            width={ props.width }
-            colorScale={ props.colorScale }
-            themeColor={ ChartThemeColor.multiOrdered }
-        >
-            <ChartAxis />
-            <ChartAxis />
-            <ChartStack horizontal>
-                <ChartBar
-                    data={ props.data1 }
-                    labelComponent={ <ChartTooltip constrainToVisibleArea /> }
-                />
-                <ChartBar
-                    data={ props.data2 }
-                    labelComponent={ <ChartTooltip constrainToVisibleArea /> }
-                />
-                <ChartBar
-                    data={ props.data3 }
-                    labelComponent={ <ChartTooltip constrainToVisibleArea /> }
-                />
-            </ChartStack>
-        </Chart>
-    );
-};
+export class StackChart extends React.Component {
```

Comment:
why'd this change from a functional component? yah can implement `useEffect` if you need lifecycle actions (similar to didmount etc)

Reply:
good point, I had changed it to add state to determine a responsive width for the stack chart, but I see what you have done with the stack chart in advisor so i was going to open a new PR to move that chart as the stack chart template, so changed this back to functional